src/pkg/scan/all/job.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2018 The Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package all
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/goharbor/harbor/src/common"
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/jobservice/job/impl/utils"
+	v1 "github.com/goharbor/harbor/src/pkg/scan/rest/v1"
+	"github.com/pkg/errors"
+)
+
+const (
+	// The max number of the goroutines to retrieve the tags
+	maxProcessors = 25
+	// Job parameter key for the admin job ID
+	jobParamAJID = "admin_job_id"
+)
+
+// Job query the DB and Registry for all image and tags,
+// then call Harbor's API to scan each of them.
+type Job struct {
+	registryURL          string
+	secret               string
+	tokenServiceEndpoint string
+	harborAPIEndpoint    string
+	coreClient           *http.Client
+}
+
+// MaxFails implements the interface in job/Interface
+func (sa *Job) MaxFails() uint {
+	return 1
+}
+
+// ShouldRetry implements the interface in job/Interface
+func (sa *Job) ShouldRetry() bool {
+	return false
+}
+
+// Validate implements the interface in job/Interface
+func (sa *Job) Validate(params job.Parameters) error {
+	_, err := parseAJID(params)
+	if err != nil {
+		return errors.Wrap(err, "job validation: scan all job")
+	}
+
+	return nil
+}
+
+// Run implements the interface in job/Interface
+func (sa *Job) Run(ctx job.Context, params job.Parameters) error {
+	logger := ctx.GetLogger()
+	logger.Info("Scanning all the images in the registry")
+
+	err := sa.init(ctx)
+	if err != nil {
+		err = errors.Wrap(err, "job handler initializer : scan all job")
+		logger.Error(err)
+		return err
+	}
+
+	// No need to check error any more as it has been checked in job validation.
+	requester, _ := parseAJID(params)
+
+	// List all the repositories of registry
+	// TODO: REPLACE DAO WITH CORRESPONDING MANAGER OR CTL
+	repos, err := dao.GetRepositories()
+	if err != nil {
+		err = errors.Wrap(err, "list repositories : scan all job")
+		logger.Error(err)
+		return err
+	}
+	logger.Infof("Found %d repositories", len(repos))
```

Comment:
logger.Debugf()?

Reply:
No, use the `info` to output it to the job log